File: rabbitmq-plugin-erlang/src/hello_world.erl
-module(hello_world).

-export([hello_world/0]).

-rabbit_boot_step({?MODULE,
                   [{description, "hello world rabbitmq plugin"},
                    {mfa, {?MODULE, hello_world, []}},
                    {requires, notify_cluster}]}).

hello_world() ->
    io:fwrite("Hello World!~n", []).

File: rabbitmq-plugin-elixir/lib/hello_world.ex
defmodule RabbitMQ.HelloWorldPlugin do
  Module.register_attribute __MODULE__,
    :rabbit_boot_step,
    accumulate: true, persist: true

  @rabbit_boot_step {__MODULE__,
                     [description: "hello world rabbitmq plugin",
                      mfa: {__MODULE__, :hello_world, []},
                      requires: :notify_cluster]}

  def hello_world() do
    IO.puts("Hello World!")
  end
end
